### ISSUE TITLE
NavigationItem uses aria-current="page" which prevents it from being used in Table of Contents

### ISSUE THREAD

**Issue by joshwooding** [jpmorganchase/salt-ds#3009]
### Package name(s)

Core (@salt-ds/core)

### Package version(s)

latest

### Description

Navigation Item uses aria-current="page" which is correct for primary navigation but not ideal for a table of contents, which should use aria-current="true". 

### Expected behavior

aria-current should be able to be passed and overriden.

### Operating system

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

**Comment by joshwooding** [comment]
This can be solved with #3417
